app/src/main/java/org/sil/gatherwords/SessionActivity.java

```diff
@@ -118,18 +130,121 @@ public void save_settings_fab_pressed(View view) {
         startActivity(i);
     }
 
-    private static class InsertSessionsTask extends AsyncTask<Session, Void, Void> {
+    private static class InsertSessionsTask extends AsyncTask<Session, Void, List<Long>> {
         private SessionDao sDAO;
 
         InsertSessionsTask(AppDatabase db) {
             sDAO = db.sessionDao();
         }
 
         @Override
-        protected Void doInBackground(Session... sessions) {
-            sDAO.insertSession(sessions);
+        protected List<Long> doInBackground(Session... sessions) {
+            return sDAO.insertSession(sessions);
+        }
+
+    }
+
+    /**
+     * Async task for inserting words from the selected word list
+     */
+    private static class InsertWordsTask extends AsyncTask<Void, Double, List<Long>> {
```

Comment:
This task is logically involved as part of the process of creating a session.  I would recommend merging this into `InsertSessionsTask` to reduce the complexity of chaining code.